pkg/operation/botanist/component/extensions/worker/worker.go

```diff
@@ -133,6 +133,9 @@ func (w *worker) Deploy(ctx context.Context) error {
 func (w *worker) deploy(ctx context.Context, operation string) (extensionsv1alpha1.Object, error) {
 	var pools []extensionsv1alpha1.WorkerPool
 
+	obj := &extensionsv1alpha1.Worker{}
+	_ = w.client.Get(ctx, client.ObjectKey{Name: w.worker.Name, Namespace: w.worker.Namespace}, obj)
+
```

Comment:
I haven't written logic to handle `notFound` error because I am not sure how to distinguish b/w the case where worker resource is `created` vs `reconciled`.
In case its `created`, a `notFound` error will always be returned, which will stop flow if we return error.
@rfranzke your inputs needed.

Reply:
I can't follow - if the `Worker` resource does not exist and we receive a "not found" error then let's not return any error (why should we?) and just continue with the below code. What's the problem?

Reply:
Ok so the current code is alright then. I just wanted to confirm abt the error returning part because of [this](https://github.com/gardener/gardener/pull/5118#discussion_r763704885) suggestion by you in the last review

Reply:
The current code is not yet alright, we should still return other errors. However, if it's a "not found" error that we get here then we just ignore it and continue.

Reply:
Oh my bad again, I missed the `!` part of `if !apierrors.IsNotFound(err)` , thats why the confusion. I will implement the changes as requested. 😓

Reply:
This looks great now :)